### PR TITLE
[EFR32]  Stack overflow on MG21 boards

### LINKED ISSUE #3223: [EFR] Hardfault when starting Thread network on a MG21 board
#### Problem
With either Lock_app and Lighting_app running on a EFR32MG21 (Board 4180A) when starting Thread network on a MG21 (BRD4180A) leads to Hardfault after a serie of calloc with done by the mbedtls encryption.

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
Starting Thread network on a MG21 (BRD4180A) would cause a Hardfault after a serie of calloc.
Defines for efr32mg21 arm MCU and architecture are wrong.

 #### Summary of Changes
- Reduce CHIP_TASK and THREAD_TASK stack size to 4096 on EFR platforms
- Fix MCU and architecture defines to Cortex-M33 and armv8-m respectively for MG21 family
- Add function to start Thread with default network info on PB0 push from a factory reset state to accelerate testing and debugging

 Fixes #3223

### CODE REVIEW

File: examples/lighting-app/efr32/BUILD.gn [hunk 1]
```diff
@@ -123,7 +123,11 @@ efr32_executable("lighting_app") {
     ldscript = "${efr32_project_dir}/ldscripts/efr32-light-example-MG21.ld"
   }
 
-  ldflags = [ "-T" + rebase_path(ldscript, root_build_dir) ]
+  ldflags = [
+    "-Wl,--print-memory-usage",
```
Comment (mspang) [inline examples/lighting-app/efr32/BUILD.gn:127]: Can you move this behind a build argument

```
if (chip_print_memory_usage) {
  ldflags += [ ... ]
}
```

Console output should be an opt-in (or written to a file) to keep signal/noise high.

File: examples/platform/efr32/Service.cpp [hunk 2]
```diff
@@ -106,3 +106,32 @@ void PublishService()
     }
     ThreadStackMgrImpl().UnlockThreadStack();
 }
+
+void StartDefaultThreadNetwork(void)
+{
+    // Set default thread network Info and enable the etWORK
```
Comment (bzbarsky-apple) [inline examples/platform/efr32/Service.cpp:112]: What is "etWORK"?

**Review by bzbarsky-apple — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

File: examples/lighting-app/efr32/BUILD.gn (hunk 1 above)
Comment (jmartinez-silabs) [reply examples/lighting-app/efr32/BUILD.gn]: Done, Thank you

File: examples/platform/efr32/Service.cpp (hunk 2 above)
Comment (jmartinez-silabs) [reply examples/platform/efr32/Service.cpp]: Must have meant network and I had a stroke ;) Fixed.

File: examples/lighting-app/efr32/BUILD.gn [hunk 3]
```diff
@@ -23,6 +23,7 @@ assert(current_os == "freertos")
 efr32_project_dir = "${chip_root}/examples/lighting-app/efr32"
 examples_plat_dir = "${chip_root}/examples/platform/efr32"
 
+chip_print_memory_usage = false
```
Comment (mspang) [inline examples/lighting-app/efr32/BUILD.gn:26]: To be able to override variables on the command line, they must be defined inside `declare_args()`.

```
declare_args() {
  # Dump memory usage at link time.
  chip_print_memory_usage = false
}
```

Comment (jmartinez-silabs) [reply examples/lighting-app/efr32/BUILD.gn]: Thank you Michael. I updated the files

**Review by mspang — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**